### PR TITLE
tests: add missing HTTP API tests for dataset management endpoints removed in #14222

### PR DESCRIPTION
**Pull request by euvre**
### Summary

PR #14222 consolidated KB (web) API endpoints into RESTful Dataset (HTTP) API endpoints and deleted the web API test suite under `test_web_api/test_kb_app/` and `test_web_api/test_document_app/`. While most test coverage was migrated to the HTTP API test suite, some tests were not ported over. This PR adds back the missing coverage.

### Route migration reference

| Old Web API | New HTTP API | Missing tests |
|---|---|---|
| `POST /v1/kb/update_metadata_setting` | `PUT /api/v1/datasets/<id>/metadata/config` | auth & error paths |
| `GET /api/v1/datasets/<id>/auto_metadata` | `GET /api/v1/datasets/<id>/metadata/config` | auth & CRUD |
| `PUT /api/v1/datasets/<id>/auto_metadata` | `PUT /api/v1/datasets/<id>/metadata/config` | auth & CRUD |
| `GET /v1/kb/<kb_id>/basic_info` | `GET /api/v1/datasets/<id>/ingestions/summary` | covered |
| `POST /v1/kb/list_pipeline_logs` | `GET /api/v1/datasets/<id>/ingestions` | edge cases missing |

#### `test_file_management_within_dataset/test_metadata_config.py` (new, 10 tests)

Covers `GET/PUT /datasets/<id>/metadata/config` (migrated from `test_kb_tags_meta.py`'s `test_update_metadata_setting` and `test_document_metadata.py`'s negative tests):
- Authorization for dataset metadata config GET/PUT
- Authorization for document metadata config PUT
- Success, invalid dataset, missing payload, not found scenarios

#### `test_dataset_management/test_ingestion_logs.py` (extended, +2 tests)

Covers `GET /datasets/<id>/ingestions` edge cases (migrated from `test_kb_pipeline_tasks.py`):
- Missing dataset ID
- Abnormal date filter

### Type of change

- [x] Other: Test coverage improvement

### CODE REVIEW

File: test/testcases/test_http_api/test_dataset_management/test_ingestion_logs.py [hunk 1]
```diff
@@ -51,3 +51,21 @@ def test_get_ingestion_log_not_found(self, HttpApiAuth, add_dataset_func):
     def test_get_ingestion_log_invalid_dataset(self, HttpApiAuth):
         res = get_ingestion_log(HttpApiAuth, "invalid_id", "some_log_id")
         assert res["code"] != 0, res
+
+
+@pytest.mark.usefixtures("clear_datasets")
+class TestListIngestionLogsEdgeCases:
+    @pytest.mark.p3
```
Comment (xugangqiang) [inline test/testcases/test_http_api/test_dataset_management/test_ingestion_logs.py:58]: By default p3 will not be executed.  Please change to p2 if want to execute it.

Comment (euvre) [reply test/testcases/test_http_api/test_dataset_management/test_ingestion_logs.py:58]: Because it was originally a p3-level test.

**Comment by xugangqiang** [comment]
LGTM
